receiver/fluentforwardreceiver/conversion.go

```diff
@@ -160,25 +162,53 @@ func parseRecordToLogRecord(dc *msgp.Reader, lr pdata.LogRecord) error {
 			return msgp.WrapError(err, "Record", key)
 		}
 
-		// fluentd uses message, fluentbit log.
-		if key == "message" || key == "log" {
-			switch v := val.(type) {
-			case string:
-				lr.Body().SetStringVal(v)
-			case []uint8:
-				// Sometimes strings come in as uint8's.
-				lr.Body().SetStringVal(string(v))
-			default:
-				return fmt.Errorf("cannot convert message type %T to string", val)
-			}
+		s, err := ValueToString(val)
+		if err != nil {
+			return err
+		}
+
+		if Contains(conf.Mappings.Body, key) {
+			insertToAttributeMap(key, val, &bodyMap)
+		} else if Contains(conf.Mappings.Severity, key) {
+			lr.SetSeverityText(s)
 		} else {
 			insertToAttributeMap(key, val, &attrs)
 		}
 	}
+	if conf.BodyAsString {
+		lr.Body().SetStringVal(lr.Body().AsString())
```

Comment:
If the user prefers a string body, we convert it here.

This will change a little the actual behavior, for a basic use case where you convert the `log` field to body we will have something like `{"log": "raw string"}` (really not sure about the output format of `lr.Body().AsString()`) instead of `raw string`

Any concern about that or the encoding choice ?